src/Airship/Internal/Decision.hs

```diff
@@ -131,6 +131,32 @@ writeCacheTags Resource{..} = lift $ do
        Nothing -> return ()
        Just d  -> addResponseHeader ("Last-Modified", utcTimeToRfc1123 d)
 
+-- | Immediately halts processing with the provided 'Status' code. The response body is
+-- | optionally set based on the error responses provided by the application and resource.
+-- | The contents of the 'Webmachine''s response body will be streamed back to the client.
+-- | This is a shortcut for constructing a 'Response' with 'getResponseHeaders' and 'getResponseBody'
+-- | and passing that response to 'finishWith'.
+haltWithErrorResponse :: Monad m => HTTP.Status -> Resource m -> Webmachine m a
+haltWithErrorResponse status Resource{..} = do
```

Comment:
Yes, I had a similar thought about whether or not this was a case that my be of interest/concern to users. I like idea of checking for a response body and only falling back to the resource's default error responses if it is not set. 

The case where someone has defined an error response for a particular status code and then wants to return a response with no body at some point in a resource seems to me like enough of an edge case that I can live with making the path to accomplish that to not define an error response for that status code for the resource and then the user must set the response body manually at the points in their resource code where they would like it returned. 

I do some playing around with this and see what comes of it.